CRM/Event/BAO/Event.php

```diff
@@ -2112,20 +2112,18 @@ public static function checkPermission(int $eventId, $permissionType = CRM_Core_
    */
   private static function checkPermissionGetInfo($eventId = NULL) {
     $params = [
-      'check_permissions' => 1,
-      'return' => 'id, created_id',
-      'options' => ['limit' => 0],
+      'select' => ['id', 'created_id'],
     ];
     if ($eventId) {
-      $params['id'] = $eventId;
+      $params['where'] = [['id', '=', $eventId]];
     }
 
     $allEvents = [];
     $createdEvents = [];
-    $eventResult = civicrm_api3('Event', 'get', $params);
-    if ($eventResult['count'] > 0) {
+    if ($eventResult = civicrm_api4('Event', 'get', $params)) {
```

Comment:
As a code-style issue, we tend to avoid putting assignments in conditionals. The previous code followed that practice.

Reply:
Also, since `$eventResult` is an APIv4 `Result` object, I'm not sure casting it to boolean would do what you expect here.
It looks to me like the conditional can just be removed, as iterating an empty `Result` set will loop 0 times, doing nothing.